containers/src/Data/IntMap.hs

```diff
@@ -70,32 +57,36 @@ module Data.IntMap
 #endif
     ) where
 
+import Prelude hiding (foldr)
+import qualified Data.IntMap.Strict as Strict
 import Data.IntMap.Lazy
 
-#ifdef __GLASGOW_HASKELL__
-import Utils.Containers.Internal.TypeError
-
--- | This function is being removed and is no longer usable.
--- Use 'Data.IntMap.Strict.insertWith'
-insertWith' :: Whoops "Data.IntMap.insertWith' is gone. Use Data.IntMap.Strict.insertWith."
-            => (a -> a -> a) -> Key -> a -> IntMap a -> IntMap a
-insertWith' _ _ _ _ = undefined
+-- | /O(log n)/. Same as 'insertWith', but the result of the combining function
+-- is evaluated to WHNF before inserted to the map.
+{-# DEPRECATED insertWith' "As of version 0.5, replaced by 'Data.IntMap.Strict.insertWith'." #-}
+{-# INLINE insertWith' #-}
+insertWith' :: (a -> a -> a) -> Key -> a -> IntMap a -> IntMap a
+insertWith' = Strict.insertWith
 
--- | This function is being removed and is no longer usable.
--- Use 'Data.IntMap.Strict.insertWithKey'.
-insertWithKey' :: Whoops "Data.IntMap.insertWithKey' is gone. Use Data.IntMap.Strict.insertWithKey."
-               => (Key -> a -> a -> a) -> Key -> a -> IntMap a -> IntMap a
-insertWithKey' _ _ _ _ = undefined
+-- | /O(log n)/. Same as 'insertWithKey', but the result of the combining
+-- function is evaluated to WHNF before inserted to the map.
+{-# DEPRECATED insertWithKey' "As of version 0.5, replaced by 'Data.IntMap.Strict.insertWithKey'." #-}
+{-# INLINE insertWithKey' #-}
+insertWithKey' :: (Key -> a -> a -> a) -> Key -> a -> IntMap a -> IntMap a
+insertWithKey' = Strict.insertWithKey
```

Comment:
These definitions should be left alone in this PR. I'm going to delete them all anyway.